pkg/models/auth/authenticator.go

```diff
@@ -67,22 +68,22 @@ type userGetter struct {
 }
 
 func NewPasswordAuthenticator(ksClient kubesphere.Interface,
-	userLister iamv1alpha2listers.UserLister,
+	ksInformer informers.SharedInformerFactory,
 	options *authoptions.AuthenticationOptions) PasswordAuthenticator {
 	passwordAuthenticator := &passwordAuthenticator{
 		ksClient:    ksClient,
-		userGetter:  &userGetter{userLister: userLister},
+		userGetter:  &userGetter{userLister: ksInformer.Iam().V1alpha2().Users().Lister()},
 		authOptions: options,
 	}
 	return passwordAuthenticator
 }
 
-func NewOAuth2Authenticator(ksClient kubesphere.Interface,
-	userLister iamv1alpha2listers.UserLister,
-	options *authoptions.AuthenticationOptions) OAuth2Authenticator {
+func NewOAuthAuthenticator(ksClient kubesphere.Interface,
+	ksInformer informers.SharedInformerFactory,
```

Comment:
Why change to SharedInformerFactory since only UserList used here

Reply:
Just like we don’t use `iamv1alpha2.IamV1alpha2Interface` directly, I think `kubesphere.SharedInformerFactory` and `kubesphere.Interface` are an integral part, and using `iamv1alpha2.UserLister` directly will make the code very complicated, right?

Reply:
not like that. sharedinfomerfactory is very big, you just use a small part of it, it means somewhere else has to create such a big instance for you. Also it's not friendly for unit testing

Reply:
fixed